File: UnitTests/BerlinClockModelTests.cs
using BerlinClock.Classes;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BerlinClock.UnitTests
{
    [TestClass]
    public class BerlinClockModelTests
    {

        [TestMethod]
        public void has_berlin_clock_second_light_off_at_23_59_59()
        {
            //arrange
            Time time = new Time(23, 59, 59);

            //act
            var clock = new BerlinClockModel(time);

            //assert
            Assert.IsFalse(clock.OnSecondLight);
        }

        [TestMethod]
        public void has_berlin_clock_second_light_on_at_23_59_58()
        {
            //arrange
            Time time = new Time(23, 59, 59);

            //act
            var clock = new BerlinClockModel(time);

            //assert
            Assert.IsFalse(clock.OnSecondLight);
        }

        [TestMethod]
        public void has_berlin_clock_4_and_3_hour_lights_on_at_23_59_59()
        {
            //arrange
            Time time = new Time(23, 59, 59);

            //act
            var clock = new BerlinClockModel(time);

            //assert
            Assert.AreEqual(4, clock.OnHourFirstRowLights);
            Assert.AreEqual(0, clock.OffHourFirstRowLights);
            Assert.AreEqual(3, clock.OnHourSecondRowLights);
            Assert.AreEqual(1, clock.OffHourSecondRowLights);
        }

        [TestMethod]
        public void has_berlin_clock_2_and_1_hour_lights_on_at_12_31_59()
        {
            //arrange
            Time time = new Time(12, 31, 59);

            //act
            var clock = new BerlinClockModel(time);

            //assert
            Assert.AreEqual(2, clock.OnHourFirstRowLights);
            Assert.AreEqual(2, clock.OffHourFirstRowLights);
            Assert.AreEqual(2, clock.OnHourSecondRowLights);
            Assert.AreEqual(2, clock.OffHourSecondRowLights);
        }

        [TestMethod]
        public void has_berlin_clock_4_and_4_minutes_lights_on_at_10_24_59()
        {
            //arrange
            Time time = new Time(10, 24, 59);

            //act
            var clock = new BerlinClockModel(time);

            //assert
            Assert.AreEqual(4, clock.OnMinutesFirstRowLights);
            Assert.AreEqual(7, clock.OffMinutesFirstRowLights);
            Assert.AreEqual(4, clock.OnMinutesSecondRowLights);
            Assert.AreEqual(0, clock.OffMinutesSecondRowLights);
        }

        [TestMethod]
        public void has_berlin_clock_0_and_0_minutes_lights_on_at_10_00_00()
        {
            //arrange
            Time time = new Time(10, 0, 0);

            //act
            var clock = new BerlinClockModel(time);

            //assert
            Assert.AreEqual(0, clock.OnMinutesFirstRowLights);
            Assert.AreEqual(11, clock.OffMinutesFirstRowLights);
            Assert.AreEqual(0, clock.OnMinutesSecondRowLights);
            Assert.AreEqual(4, clock.OffMinutesSecondRowLights);
        }

    }
}

File: BerlinClock.UnitTests/BerlinClockModelTests.fs
module BerlinClockModelTests

open NUnit.Framework
open System
open BerlinClock.Classes

[<Test>]
let ``has berlin clock second light off at 23:59:59``() = 
    //arrange
    let time = new Time(23, 59, 59);

    //act
    let clock = new BerlinClockModel(time);

    //assert
    Assert.That(clock.OnSecondLight, Is.False);

[<Test>]
let ``has berlin clock second light on at 23:59:58``() = 
    //arrange
    let time = new Time(23, 59, 59);

    //act
    let clock = new BerlinClockModel(time);

    //assert
    Assert.That(clock.OnSecondLight, Is.False);

[<Test>]
let ``has berlin clock 4 and 3 hour lights on at 23:59:59``() = 
    //arrange
    let time = new Time(23, 59, 59);

    //act
    let clock = new BerlinClockModel(time);

    //assert
    Assert.That(clock.OnHourFirstRowLights, Is.EqualTo(4));
    Assert.That(clock.OffHourFirstRowLights, Is.EqualTo(0));
    Assert.That(clock.OnHourSecondRowLights, Is.EqualTo(3));
    Assert.That(clock.OffHourSecondRowLights, Is.EqualTo(1));

[<Test>]
let ``has berlin clock 2 and 1 hour lights on at 12:31:59``() = 
    //arrange
    let time = new Time(12, 31, 59);

    //act
    let clock = new BerlinClockModel(time);

    //assert
    Assert.That(clock.OnHourFirstRowLights, Is.EqualTo(2));
    Assert.That(clock.OffHourFirstRowLights, Is.EqualTo(2));
    Assert.That(clock.OnHourSecondRowLights, Is.EqualTo(2));
    Assert.That(clock.OffHourSecondRowLights, Is.EqualTo(2));

[<Test>]
let ``has berlin clock 4 and 4 minutes lights on at 10:24:59``() = 
    //arrange
    let time = new Time(10, 24, 59);

    //act
    let clock = new BerlinClockModel(time);

    //assert
    Assert.That(clock.OnMinutesFirstRowLights, Is.EqualTo(4));
    Assert.That(clock.OffMinutesFirstRowLights, Is.EqualTo(7));
    Assert.That(clock.OnMinutesSecondRowLights, Is.EqualTo(4));
    Assert.That(clock.OffMinutesSecondRowLights, Is.EqualTo(0));

[<Test>]
let ``has berlin clock 0 and 0 minutes lights on at 10:00:00``() = 
    //arrange
    let time = new Time(10, 0, 0);

    //act
    let clock = new BerlinClockModel(time);

    //assert
    Assert.That(clock.OnMinutesFirstRowLights, Is.EqualTo(0));
    Assert.That(clock.OffMinutesFirstRowLights, Is.EqualTo(11));
    Assert.That(clock.OnMinutesSecondRowLights, Is.EqualTo(0));
    Assert.That(clock.OffMinutesSecondRowLights, Is.EqualTo(4));
